### DISCUSSION TITLE
The backward compatibility of Istio 1.22 with Istio 1.20 doesn't work

### DISCUSSION THREAD

**Discussion by angroszew** [istio/istio#53655] [Q&A] [answered]
Hello,
I upgraded Istio from 1.20.3 to 1.22.2. The upgrade successfully finished. 
I have several applications in one namespace in Istio mesh that were able to communicate with each other before the Istio upgrade, for example keycloak application was able to connect with MariaDB.
After the Istio upgrade I restarted MariaDB pods executing `kubectl rollout restart` command. MariaDB pods successfully restarted and istio-proxy containers started using Istio 1.22.2 image. But after MariaDB was successfully restarted, Keycloak application (with istio-proxy 1.20.3) wasn't able to communicate with MariaDB. When I executed `istioctl proxy-status` only MariaDB pods were reported in the output. There were no Keycloak pods in Istio mesh. I found the following errors in istiod, discovery logs: 
`2024-10-25T08:47:09.346809Z     error   security        Failed to authenticate client from 192.168.120.127:37392: Authenticator ClientCertAuthenticator: no verified chain is found; Authenticator KubeJWTAuthenticator: failed to validate the JWT from cluster "Kubernetes": the service account authentication returns an error: [invalid bearer token, token audiences ["https://kubernetes.default.svc.cluster.local"] is invalid for the target audiences ["istio-ca"]]`
I had to execute `kubectl rollout restart` also for Keycloak to restore connectivity from Keycloak to MariaDB.

Is it a bug or an intended behavior that pods with istio-proxy 1.20.3 cannot be added to Istio 1.22.2 mesh and cannot communicate with applications that have istio-proxies with Istio 1.22.2?

I would be grateful for any tips and advice.

Best Regards,
Ania

**Comment by howardjohn** [comment, marked as answer]
This sounds like usage of `JWT_POLICY=first-party-jwt` or similar, an option that was deprecated ~5 years ago in Istio 1.2 that was finally removed

**Comment by angroszew** [comment]
@howardjohn,
Is upgrade from Istio 1.20 (with 1st party jwt) to Istio 1.22 (with 3rd party jwt) possible with NO impact on the connections between applications? I did such upgrade but I had to execute `kubectl rollout restart` after the upgrade to 1.22 for each application to force picking it to the latest proxy.
